Adding Corsican translation to GeneWeb

Patriccollu opened the pull request:
Hi GeneWeb team,

This is my first post and I hope it's the right place.

I appreciate GeneWeb and I would like to contribute to this project by adding a new translation to the interface. That's why I started to translate the interface in **Corsican**!

The translation has been based on versions 6.08 and 7.00a. So far I updated 4 files - lex_utf8.txt, start_utf8.txt, version.txt, welcome.htm - and created 2 new icons: l-co.jpg and t-co.png. Please tell me if any other file needs to be translated?

I would be more than happy if you could include Corsican in the next distributed version!

Best regards,
Patriccollu di Santa Maria è Sichè

Patriccollu committed:
Updating lex_utf8.txt to add Corsican translations

Patriccollu committed:
Adding l-co.jpg Corsican flag (40 x 25)

Patriccollu committed:
Adding t-co.png

co in green letters, icon with transparency (36 x 22)

Patriccollu committed:
Updating start_utf8.txt to add Corsican translations

Patriccollu committed:
Updating version.txt to add Corsican translations

Patriccollu committed:
Updating welcome.htm to add Corsican translations

Patriccollu committed:
Update welcome.htm

Patriccollu committed:
Updating start_utf8.txt to correct some errors

Patriccollu committed:
Updating lex_utf8.txt to correct some errors

Updating lex_utf8.txt to correct some errors and remove extra lines

a2line commented:
Hi. Thanks for your great contribution.

Can you work a bit more on the “!languages” block that is in `lex-utf8` and add the names of the Corsican language after Czech to all other languages? For eg. with `/co=corse` for `fr:` line:
```
fr: af=afrikaans/bg=bulgare/br=breton/ca=catalan/cs=tchèque/co=corse/da=danois/de=allemand/en=anglais/eo=espéranto/es=espagnol/et=estonien/fi=finnois/fr=français/he=hébreu/is=islandais/it=italien/lv=letton/nl=néerlandais/no=norvégien/oc=occitan/pl=polonais/pt=portugais/pt-br=portugais du Brésil/ro=roumain/ru=russe/sk=slovaque/sl=slovène/sv=suédois/zh=chinois
```

You should also add a line for Corsican to the button at the bottom of pages that allow to switch languages, template file [copyr.txt line 47](https://github.com/geneweb/geneweb/blob/master/hd/etc/copyr.txt#L47) (non template pages). For template pages, if you have corrected !languages block in `lex-utf8`, on `copyr.txt` add a `/co` to [line 4](https://github.com/geneweb/geneweb/blob/master/hd/etc/copyr.txt#L4) (after `/cs`), and on [line 8](https://github.com/geneweb/geneweb/blob/master/hd/etc/copyr.txt#L8) switch the number of languages from `30` to `31`. That way you can switch to Corsican from everywhere!

You should also add a button for Corsican adding a line after [index.txt line 34](https://github.com/geneweb/geneweb/blob/master/hd/etc/index.txt#L34) (index is first page before welcome, when we are on http://myserver:2317/ without any base selected).

fablhx commented:
Sorting the `lex_utf8` file may be done automatically if this is a problem.

Patriccollu commented:
Regarding the alphabetical order, Corsican language should be placed after Catalan and **before** Czech.

`ca/co/cs`

Patriccollu committed:
Updating lex-utf8 to add the names of the Corsican language

Updating lex-utf8 to add the names of the Corsican language on the “!languages” block.

Patriccollu committed:
Updating copyr.txt to add Corsican language

Patriccollu committed:
Updating index.txt to add Corsican language

Patriccollu commented:
Hello,

I just updated lex-utf8.txt, copyr.txt, and index.txt. Is it what you were expecting?

Thanks,
Patriccollu di Santa Maria è Sichè

a2line commented:
If it's working for you, the form looks good from here.

Patriccollu commented:
@a2line 
Hello,

Please can you tell me - if it's already scheduled - when the next version of GeneWeb will include new Corsican language?

I've also noticed another GeneWeb repository on `github.com/geneanet/geneweb` with the same contributors, me excepted! ;-)

Do I need to deliver it some modifications similar to what I made in the present Pull Request?

By the way, what's the difference between both projects/repositories?

Thanks,
Patriccollu di Santa Maria è Sichè

sagotch commented:
Hello.

You do not need to submit your change to geneanet's repo, they will be merge when master branch of `geneanet/geneweb` will rebase on `geneweb/geneweb`.

Difference between `geneanet/geneweb` and `geneweb/geneweb` is the same as between `Patriccollu/geneweb` and `geneweb/geneweb`. Both `Patriccollu` and `geneanet` are forks.

Next release of geneweb is likely to include corsican language since it is in master, which will probably be the branch used for the next release.

Patriccollu commented:
@a2line 
Hello,

I noticed that Corsican language can now be selected on the GeneWeb 7.00 online demo base:

	http://demo.geneweb.tuxfamily.org/gw7/gwd?b=grimaldi&lang=co;

Unfortunately, when checking the Download page - https://geneweb.tuxfamily.org/wiki/download - the latest versions for Windows - 6.08 and 7.00 alpha - have not recently been updated!

Do you know when the next version of GeneWeb - which will include new Corsican language - is planned to be released?

Thanks,
Patriccollu di Santa Maria è Sichè.

sagotch commented:
Je crois qu'une prochaine (pre)version est prévue après la fusion des branches:
- https://github.com/geneweb/geneweb/pull/711 (je suis en train de travailler à son adaptation pour pouvoir fusionner)
- https://github.com/geneweb/geneweb/pull/742
- https://github.com/geneweb/geneweb/pull/744
- Un paquet de correctifs que je pousserai une fois le reste incorporé dans master

Patriccollu commented:
@sagotch 
Merci pour l'information.
